### DISCUSSION TITLE
Quarto installation in a corporate environnement ?

### DISCUSSION THREAD

**Discussion by lcrmorin** [quarto-dev/quarto-cli#6485] [Q&A]
### Description

I have the occasion to use quarto in a personnal setting, where I can install and use whatever I want. However, in my professional environnement I am not able to do so. I was wondering if someone has experience making it work in a big corporate environnement that don't really allow installing external programs / extensions. Do you necessarily have to go the long route of withelisting it somewhere ? or is there any alternative ?

**Comment by mcanouil** [comment]
We can't tell you to bypass your corporate security rules, but Quarto does not require admin rights to be installed.

**Comment by cscheid** [comment]
One possible alternative is for you to become comfortable with continuous integration. You could then use github workflows to produce your document outputs. We have a separate repository with documentation and examples for such setups: https://github.com/quarto-dev/quarto-actions. It takes a bit of learning, but the result is that you can run whatever version of quarto you want, and rendering happens on GitHub's CI servers.

**Reply by lcrmorin** [reply]
Thanks, ths is an interesting idea. I'll try this with my personnal GitHub. I am afraid this might not be easy / possible to share corporate information with github. I'll try with our internal Git/Gitlab.

**Comment by harrelfe** [comment]
IMHO `Quarto` is advancing too rapidly and too many minor bug corrections are being made for a "frozen" version of `Quarto` to be useful.  That opinion implies that `Quarto` needs to be installed and updated on user machines unless the corporate IT guru agrees to update `Quarto` nightly.

**Reply by mcanouil** [reply]
If you stick to the stable version, you don't have that much updates as patch releases are released every few months and not for "minor bugs"[^1].
Of course, if you are using the pre-release builds then you have to update really frequently.

[^1]: e.g., there is an R release every trimester.

**Reply by cscheid** [reply]
> IMHO Quarto is advancing too rapidly and too many minor bug corrections are being made for a "frozen" version of Quarto to be useful. That opinion implies that Quarto needs to be installed and updated on user machines unless the corporate IT guru agrees to update Quarto nightly.

That's a fair assessment, but I'll reiterate my previous point about CI: if one is willing to learn about GitHub actions, then it's easy to configure those to either use a specific version or to always use the latest pre-release.

**Comment by dpastoor** [comment]
I have addressed this challenge mostly in https://github.com/dpastoor/qvm _given_ your environment you do have egress to the internet. If you are air gapped and the blocker is literally even pulling the installation information in it requires more hoops.

I am currently using this in admin'ing validated platforms in pharma from top 10 pharma to small biotech.

See the release notes: https://github.com/dpastoor/qvm/releases/tag/v0.3.0 there are steps as an "admin" but also as a single user. 

<img width="1232" alt="image" src="https://github.com/quarto-dev/quarto-cli/assets/3196313/6ac72624-72a4-48d1-a736-52f12ac1eaa3">

In this case, the single user flow keeps everything about quarto scoped to your home directory. In addition it allows you to switch between quarto versions for testing/reproducibility etc.

This also works well in containerized environments like kubernetes where your homedir is dynamically mounted across containers and quarto is not baked into the containers themselves.
